### PR TITLE
ENH: Refactor tractography episodes

### PR DESCRIPTION
**Pull request by jhlegarreta**
Refactor tractography episodes:
- Add the deterministic tracking episode Markdown file.
- Make the deterministic tractography be consistent with the dataset and subject
  id used in the current version of the lesson.
- Use `matplotlib` to display the tractogram scene in the deterministic
  tractography episode and avoid opening the `fury` window.
- Add contents to the local tracking Markdown file.
- Improve the explanations in the probabilistic tracking episode.
- Refactor the tractography general Markdown file to make the contents more
  consistent across the related episodes.
- Remove the output error messages in the deterministic tracking Jupyter
  notebook.

Take advantage of the commit to:
- Remove the `{: .language-python}` episode Markdown-specific markup from the
  probabilistic tracking Jupyter notebook.
- Remove the episode Markdown-specific header from the probabilistic tracking
  Jupyter notebook.
- Add the subject id to the probabilistic tracking Markdown file and remove the
  unnecessary/unused call to `window.record`.
- Make the episode titles be consistent.
- Use the reference to the first work that proposed the global tracking method.
- Fix typos (e.g. `propogate` to `propagate`; `see density` to `seed density`).
- Use the term "episode" to refer to the sections.
- Use a double white space before inline comments to conform to PEP8.

### CODE REVIEW

**Review by kaitj — CHANGES_REQUESTED** [review]
Looks good, a couple of things should be addressed and I've added comments to those lines.

Not sure if this was discussed, but maybe we should think about updating the file names to match the episode number? For example, `08-deterministic_tractography.md` corresponds to episode 4.2.

File: _episodes/08-deterministic_tractography.md [hunk 1]
```diff
@@ -1,13 +1,217 @@
...
+~~~
+from dipy.io.stateful_tractogram import Space, StatefulTractogram
+from dipy.io.streamline import save_trk
+
+sft = StatefulTractogram(streamlines, dwi_img, Space.RASMM)
+save_trk(sft, "tractogram_EuDX.trk")
+~~~
+{: .language-python}
+
+We can then display the resulting streamlines using the `fury` python package
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply _episodes/08-deterministic_tractography.md]: Re-worded in 13fdeec.

File: _episodes/08-deterministic_tractography.md [hunk 2]
```diff
@@ -1,13 +1,217 @@
...
+
+We will determine the indices representing the discretized directions of the
+peaks by providing as input, our tensor model, the diffusion data, the sphere,
+and a mask to apply the processing to. Additionally, we will set the minimum
+angle between directions, the maximum number of peaks to return (1 for the
+tensor model), and the relative peak threshold (returning peaks greater than
+this value).
+
+> Note
+> This step may take a while to run_
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply _episodes/08-deterministic_tractography.md]: It was a leftover from the italics syntax from the notebook. Addressed in faf8ff5.

**Comment by jhlegarreta** [comment]
> Not sure if this was discussed, but maybe we should think about updating the file names to match the episode number? For example, 08-deterministic_tractography.md corresponds to episode 4.2.

Yep, we have mentioned this before (e.g. [here](https://github.com/carpentries-incubator/SDC-BIDS-dMRI/pull/47#issuecomment-762560492) and [here](https://github.com/carpentries-incubator/SDC-BIDS-dMRI/pull/52#issuecomment-766915476)). Rather than updating the names to match the episode number, I'd rename the files to partially match the episode content/title. That way, when a new episode has to be added somewhere in the middle we would not need to rename all subsequent files.

Whether we do it one way or the other, I'd do it in a separate PR to push things forward.

File: _episodes/08-deterministic_tractography.md [hunk 3]
```diff
@@ -1,13 +1,218 @@
...
-- "FIXME"
+- "What computations does a deterministic tractography require?"
+- "How can we visualize the streamlines generated by a tractography method?"
 objectives:
-- "FIXME"
+- "Be able to perform deterministic tracking on diffusion MRI data"
+- "Familiarize with the data entities of a tractogram"
 keypoints:
-- "FIXME"
+- "Deterministric tractography methods perform tracking in a predictable way"
```
Comment (kaitj) [inline _episodes/08-deterministic_tractography.md:12]: Spelling here - "Deterministic"

**Review by kaitj — COMMENTED** [review]
Looks good, just one minor spelling correction in `08-deterministic_tractography.md`

File: _episodes/08-deterministic_tractography.md (hunk 3 above)
Comment (jhlegarreta) [reply _episodes/08-deterministic_tractography.md]: :+1: Fixed in 7b15bf5.

**Comment by josephmje** [comment]
Thanks @jhlegarreta! I don't know as much about tractography so I'll let @kaitj provide his input here. I'm happy to take a stab at working through these lessons once they're merged to see if I can help with clarity as a "newish" learner.

Also, looks like the website previews are finally working!

**Comment by jhlegarreta** [comment]
> I'm happy to take a stab at working through these lessons once they're merged to see if I can help with clarity as a "newish" learner.

Thanks Michael. Yes, a few parts will require some changes throughout the lesson, but the essential/core aspects/content is there. IMHO the lesson is getting close to being stable.

> Also, looks like the website previews are finally working!

Yes, I saw that. Wonderful, extremely useful :100: ! Thanks for doing that.

File: _episodes/09-probabilistic_tractography.md [hunk 4]
```diff
@@ -9,27 +9,18 @@ objectives:
...
+- "Provides tractograms that explore more white matter axonal fibers"
 ---
 
-## Probabilistic Tractography
+## Probabilistic tractography
 
 Probabilistic fiber tracking is a way of reconstructing the white matter
 structural connectivity using diffusion MRI data. Much like deterministic fiber
 tracking, the probabilistic approach follows the trajectory of a possible
 pathway in a step-wise fashion and propagating streamlines based on the local
```
Comment (kaitj) [inline _episodes/09-probabilistic_tractography.md:21]: would change "propagating streamlines based on the local..." to "propagate streamlines based on the local..."

**Review by kaitj — COMMENTED** [review]
Just a couple more minor things, otherwise it looks good. 

The example output shown (line 316+ in `09-probabilistic tractography.md`) looks good in an IDE / on git (probably as this what is shown in the terminal), but is hard to read on the rendered site. I wonder if it would be helpful to put in a code block? Can be done in another PR.

File: _episodes/09-probabilistic_tractography.md (hunk 4 above)
Comment (jhlegarreta) [reply _episodes/09-probabilistic_tractography.md]: Changed in f6bd4c9. Force pushed.

**Comment by jhlegarreta** [comment]
> The example output shown (line 316+ in 09-probabilistic tractography.md) looks good in an IDE / on git (probably as this what is shown in the terminal), but is hard to read on the rendered site. I wonder if it would be helpful to put in a code block? Can be done in another PR.

You're right, it does not look nice on the website. Needs definitely to be reworked. Best would be a separate PR.

**Comment by jhlegarreta** [comment]
Ready to be merged.

**Review by kaitj — APPROVED** [review]
Looks good! And yeah we can do a separate PR for the code output.

**Pull request merged**